qiskit/validation/fields/custom.py

```diff
@@ -47,12 +47,11 @@ def _serialize(self, value, attr, obj, **_):
             raise self.make_error_serialize('format', input=value)
 
     def _deserialize(self, value, attr, data, **_):
-        if not is_collection(value) or len(value) != 2:
-            raise self.make_error('invalid', input=value)
-
-        try:
-            return complex(*value)
-        except (ValueError, TypeError):
+        if is_collection(value) and (len(value) == 2):
+            return complex(value[0], value[1])
```

Comment:
Do you think it's worth wrapping this in try except like before:

```python
try:
    return complex(value[0], value[1])
except (ValueError, TypeError):
    raise self.make_error('invalid', input=value)
```

Right now we'll get a ValueError or TypeError, but before we'd get a marshmallow exception. I'm not sure it makes a big difference though.

Reply:
I wasn't sure about this actually - shouldn't the collection and len checks make Value/Type errors really unlikely? 

As long as it doesn't affect performance (which I don't think it will) I can add it back in - I wanted to remove the else line anyway

Reply:
My assumption this was a check for something like an input with something like:

value = ['real', 'imaginary']

which would match the if check but not be valid input to cast. That being said this just changes the error case and the exception raised. It shouldn't have any performance overhead.

Reply:
very true, there aren't any checks on the types within the collection - I'll add the try except as you suggested

Reply:
fixed in 0ad5da7